src/main/java/io/jenkins/plugins/artifact_manager_jclouds/s3/S3BlobStore.java

```diff
@@ -137,17 +151,32 @@ public BlobStoreContext getContext() throws IOException {
      */
     private Supplier<Credentials> getCredentialsSupplier() throws IOException {
         // get user credentials from env vars, profiles,...
-        AmazonS3ClientBuilder builder = getConfiguration().getAmazonS3ClientBuilder();
-        AWSSessionCredentials awsCredentials = CredentialsAwsGlobalConfiguration.get().sessionCredentials(builder);
-        String sessionToken = awsCredentials.getSessionToken();
+        String accessKeyId;
+        String secretKey;
+        String sessionToken;
+        
+        if (getConfiguration().getDisableSessionToken()) {
+            AmazonWebServicesCredentials awsCredentials = CredentialsAwsGlobalConfiguration.get().getCredentials();
+            accessKeyId = awsCredentials.getCredentials().getAWSAccessKeyId();
+            secretKey = awsCredentials.getCredentials().getAWSSecretKey();
+            sessionToken = "";
+        } else {
+            AmazonS3ClientBuilder builder = getConfiguration().getAmazonS3ClientBuilder();
+            AWSSessionCredentials awsCredentials = CredentialsAwsGlobalConfiguration.get().sessionCredentials(builder);
```

Comment:
I don't understand why `AWSSessionCredentials` are necessary here. As far as I understand, there are only two cases:
* Access Key ID/Secret Key are provided -> they can be provided directly.
* They are not provided, try to infer from instance profile, then you get an `AWSSessionCredentials` indeed.

The only case where potentially it could be used is if you have MFA enabled but in any case it wouldn't be applicable for this use case.

So in my opinion, this `disableSessionToken` shouldn't be an option.

Reply:
No idea, this is from the upstream PR, which documents that you must select this option to use MinIO. I will try it without and see what happens.

Reply:
The problem is that the underlying code from `sessionCredentials` calls some AWS verb to obtain temporary credentials even if it has long-term credentials.

Reply:
```
com.amazonaws.services.securitytoken.model.AWSSecurityTokenServiceException: The security token included in the request is invalid. (Service: AWSSecurityTokenService; Status Code: 403; Error Code: InvalidClientTokenId; Request ID: …)
```

solved by checking **Disable Session Token**.

Reply:
> the underlying code from `sessionCredentials` calls some AWS verb to obtain temporary credentials even if it has long-term credentials

I suppose. Anyway, unless you have some idea how to determine automatically whether or not to attempt to get session credentials, I am inclined to leave it as a configuration option which seems to work.

Reply:
I mean, we could call `CredentialsAwsGlobalConfiguration.sessionCredentials` only when there is an `accessKey` + `secretKey` but _not_ a `iamRoleArn` or `iamMfaSerialNumber`. This would potentially change the behavior of the plugin for existing users on S3, though, and feels risky and magical.